File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewmodel/clearalldata/ClearAllDataViewModel.java
package com.dewonderstruck.apps.ashx0.viewmodel.clearalldata;

import com.dewonderstruck.apps.ashx0.repository.clearpackage.ClearPackageRepository;
import com.dewonderstruck.apps.ashx0.utils.AbsentLiveData;
import com.dewonderstruck.apps.ashx0.viewmodel.common.PSViewModel;
import com.dewonderstruck.apps.ashx0.viewobject.common.Resource;

import javax.inject.Inject;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;

public class ClearAllDataViewModel extends PSViewModel {

    private final LiveData<Resource<Boolean>> deleteAllDataData;
    private MutableLiveData<Boolean> deleteAllDataObj = new MutableLiveData<>();


    @Inject
    public ClearAllDataViewModel(ClearPackageRepository repository) {

        deleteAllDataData = Transformations.switchMap(deleteAllDataObj, obj -> {
            if (obj == null) {
                return AbsentLiveData.create();
            }
            return repository.clearAllTheData();
        });
    }

    public void setDeleteAllDataObj() {

        this.deleteAllDataObj.setValue(true);
    }

    public LiveData<Resource<Boolean>> getDeleteAllDataData() {
        return deleteAllDataData;
    }


}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewmodel/clearalldata/ClearAllDataViewModel.kt
package com.dewonderstruck.apps.ashx0.viewmodel.clearalldata

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import com.dewonderstruck.apps.ashx0.repository.clearpackage.ClearPackageRepository
import com.dewonderstruck.apps.ashx0.utils.AbsentLiveData
import com.dewonderstruck.apps.ashx0.viewmodel.common.PSViewModel
import com.dewonderstruck.apps.ashx0.viewobject.common.Resource
import javax.inject.Inject

class ClearAllDataViewModel @Inject constructor(repository: ClearPackageRepository) : PSViewModel() {
    val deleteAllDataData: LiveData<Resource<Boolean>>
    private val deleteAllDataObj = MutableLiveData<Boolean>()
    fun setDeleteAllDataObj() {
        deleteAllDataObj.value = true
    }

    init {
        deleteAllDataData = Transformations.switchMap(deleteAllDataObj) { obj: Boolean? ->
            if (obj == null) {
                return@switchMap AbsentLiveData.create<Resource<Boolean>>()
            }
            repository.clearAllTheData()
        }
    }
}